packages/expo-updates/android/src/androidTest/java/expo/modules/updates/loader/FileDownloaderTest.kt

```diff
@@ -123,6 +123,34 @@ class FileDownloaderTest {
     Assert.assertEquals("custom", actual.header("expo-updates-environment"))
   }
 
+  @Test
+  @Throws(JSONException::class)
+  fun testAssetExtraHeaders_ObjectTypes() {
+    val configMap = mapOf<String, Any>(
+      "updateUrl" to Uri.parse("https://u.expo.dev/00000000-0000-0000-0000-000000000000"),
+      "runtimeVersion" to "1.0",
+    )
+
+    val config = UpdatesConfiguration(null, configMap)
+
+    val extraHeaders = JSONObject().apply {
+      put("expo-string", "test")
+      put("expo-number", 47.5)
+      put("expo-boolean", true)
+    }
+
+    val assetEntity = AssetEntity("test", "jpg").apply {
+      url = Uri.parse("https://example.com")
+      extraRequestHeaders = extraHeaders
+    }
+
+    // assetRequestHeaders should not be able to override preset headers
```

Comment:
```suggestion
    // assetRequestHeaders should have their values coerced to strings
```
or something to that effect